### PR TITLE
Account Detail Service

### PR DESCRIPTION
**Pull request by piyushswain**
This PR adds the `SolidusBolt::Accounts::DetailService` API Object.

This object inherits from the `SolidusBolt::Base` Service Object and allows to call the `/account` api for Bolt.

This requires a single argument which is an `access_token`.

Bolt API DOCS reference: https://help.bolt.com/api-bolt/#tag/Account/operation/GetAccountDetails

### CODE REVIEW

File: spec/services/solidus_bolt/accounts/detail_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+RSpec.describe SolidusBolt::Accounts::DetailService, :vcr, :bolt_configuration do
+  subject(:api) { described_class.new(access_token: access_token) }
+
+  let(:access_token) { 'Bolt Access Token' }
+
+  describe '#call', vcr: true do
+    it 'receives the correct response' do
+      expect{ api.call }.to raise_error(SolidusBolt::ServerError, 'This action is forbidden.')
+    end
```
Comment (DanielePalombo) [inline spec/services/solidus_bolt/accounts/detail_service_spec.rb:13]: IMO This failing test is not enough.

Considering we'll add other services which will use AccessToken, we should provide a way to specify it to the local env.

**Review by DanielePalombo — CHANGES_REQUESTED** [review]

File: spec/services/solidus_bolt/accounts/detail_service_spec.rb (hunk 1 above)
Comment (piyushswain) [reply spec/services/solidus_bolt/accounts/detail_service_spec.rb]: Modified this test here
[5a6db27](https://github.com/nebulab/solidus_bolt/commit/5a6db27026af90bed6d07db9857e40f0baca3c19)

File: spec/services/solidus_bolt/accounts/detail_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        expect(api.call).to match hash_including('payment_methods')
+      end
+    end
+  end
+end
+
+# This spec depends on the @bolt_access_token that needs to be generated manually
+# and added to the environment variable BOLT_ACCESS_TOKEN.
+# Generate a new access token by logging into an existing user bolt account on
+# development environment and copying the token from the session['bolt_access_token'] key.
```
[the remark this answers is not in the document]
Comment (piyushswain) [reply spec/services/solidus_bolt/accounts/detail_service_spec.rb:34]: We should add this information to the README but I prefer if we let this comment stay in this file as well

File: spec/services/solidus_bolt/accounts/detail_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      let(:access_token) { 'Bolt Access Token' }
+
+      it 'gives an error' do
+        expect{ api.call }.to raise_error(SolidusBolt::ServerError, 'This action is forbidden.')
+      end
+    end
+
+    context 'with correct access_token' do
+      # rubocop:disable RSpec/InstanceVariable
+      subject(:api) { described_class.new(access_token: @bolt_access_token) }
```
[the remark this answers is not in the document]
Comment (piyushswain) [reply spec/services/solidus_bolt/accounts/detail_service_spec.rb]: Modified here
[4ae5598](https://github.com/nebulab/solidus_bolt/commit/4ae5598ad6d2d7ac0c7673d78218d34fff1a2f60)

File: README.md [hunk 4]
```diff
@@ -110,6 +110,15 @@ factories along with this extension's factories using this statement:
 SolidusDevSupport::TestingSupport::Factories.load_for(SolidusBolt::Engine)
 ```
 
+#### Special Tests
+
+A few tests in the test suite require some additional steps to execute successfully when they are modified. These are listed below along with the steps needed to execute these tests.
+
+- `/spec/services/solidus_bolt/accounts/detail_service_spec.rb`
+This test requires a valid bolt access token to execute successfully. Generate the bolt access token by logging into a bolt sandbox account in development and copying the token from the `session['bolt_access_token']` key.
```
Comment (Naokimi) [inline README.md:118]: I'm not clear I understand it myself by reading this. Maybe some more details or a link would help with clarity

File: spec/services/solidus_bolt/accounts/detail_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe SolidusBolt::Accounts::DetailService, :vcr, :bolt_configuration do
+  describe '#call', vcr: true do
+    context 'with wrong access_token' do
+      subject(:api) { described_class.new(access_token: access_token) }
```
Comment (Naokimi) [inline spec/services/solidus_bolt/accounts/detail_service_spec.rb:8]: this and line 18 should be extracted to between line 6 and 7

File: README.md (hunk 4 above)
Comment (piyushswain) [reply README.md:118]: I don't really have a link for this, but could you let me know what extra details would you like in this explanation to make it more easy yo understand.

File: spec/services/solidus_bolt/accounts/detail_service_spec.rb (hunk 5 above)
Comment (piyushswain) [reply spec/services/solidus_bolt/accounts/detail_service_spec.rb]: fixed here
[4816aeb](https://github.com/nebulab/solidus_bolt/commit/4816aeb0685d59f8bd9e005454b69ef638010228)

File: README.md (hunk 4 above)
Comment (piyushswain) [reply README.md]: Modified here
[6ad4d41](https://github.com/nebulab/solidus_bolt/commit/6ad4d41739310671594d966c6a2b5c25c3edbade)

**Review by DanielePalombo — APPROVED** [review]
LGTM

**Review by Naokimi — APPROVED** [review]

**Pull request merged**